crates/ra_hir_def/src/nameres/collector.rs

```diff
@@ -552,8 +554,12 @@ where
         self.unexpanded_macros = macros;
         self.unexpanded_attribute_macros = attribute_macros;
 
-        for (module_id, macro_call_id) in resolved {
-            self.collect_macro_expansion(module_id, macro_call_id);
+        for (module_id, macro_call_id, depth) in resolved {
+            if depth > 128 {
```

Comment:
although the default rustc `recursion_limit is 128, but I think 1024 would be our [safe bet](https://grep.app/search?q=%23%21%5Brecursion_limit%20%3D%20&filter[lang][0]=Rust) here. 

OTOH, One day, we will set it via the `recursion_limit` of the  module `attr`.

Reply:
I reduced it to try to get the second test to pass in an acceptable time, but it wasn't enough anyway.

Reply:
Yeah, even [playground](https://play.rust-lang.org/) refused to run too :)